fix(notifications/manual-judgements): don't let echo work wrap slack messages to 80 chars

kevinawoo opened the pull request.

kevinawoo committed:
fix(notifications/manual-judgements): don't let echo work wrap slack messages to 80 chars

link108 commented on echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/HtmlToPlainText.java:
```diff
@@ -0,0 +1,76 @@
+package com.netflix.spinnaker.echo.notification; /*
```
package after copyright

jeffhart-armory commented on echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/HtmlToPlainText.java:
```diff
@@ -0,0 +1,76 @@
+package com.netflix.spinnaker.echo.notification; /*
```
Your copyright and import statements are a little messed up.

kevinawoo replied:
damn you gradlew... what are you doing!?

link108 commented on echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/HtmlToPlainText.java:
```diff
@@ -0,0 +1,76 @@
+      String name = node.nodeName();
+      if (StringUtil.in(name, "br", "dd", "dt", "p", "h1", "h2", "h3", "h4", "h5")) {
+        append("\n");
+      } else if (name.equals("a")) {
```
should this if/else be reversed? I don't see a default case, so am wondering if it'd be better to do:

```
if (name.equals("a")) {
  append(String.format(" <%s>", node.absUrl("href")));
} else {
   append("\n");
}
```
but you may want to skip everything that doesn't match that list, so depends on what you want to do

kevinawoo replied:
🤷‍♂ 	copy and paste job from https://github.com/jhy/jsoup/blob/jsoup-1.8.2/src/main/java/org/jsoup/examples/HtmlToPlainText.java

kevinawoo replied:
i guesssssssssss I should fix it >_<

kevinawoo replied:
oh actually, we just drop the tail, (kind of unexplicitfully)

link108 commented on echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/HtmlToPlainText.java:
```diff
@@ -0,0 +1,76 @@
+
+    // appends text to the string builder
+    private void append(String text) {
+      if (text.equals(" ")
```
may want to clean this up so its easier to read. Can do so in two ways: 

1. Make a function for the test 
2. Add parentheses so the order of operations is easier to read without thinking too hard about :)

eg:
```
if ( text_and_accum_empty || just_newlines) {
  return;
}
```

link108 replied:
ah I see the left paren on the or clause, spotless messed this up!

kevinawoo committed:
Merge branch 'master' into feature/dont-wordbreak-notifications

spinnakerbot committed:
chore(dependencies): Autobump korkVersion (#615)

kevinawoo committed:
Merge branch 'master' into feature/dont-wordbreak-notifications

ezimanyi committed:
chore(build): Update cloudbuild.yaml file (#618)

Remove the $REPO_NAME variable from the cloudbuild.yaml file; this is being used
to decide the name of the image to push which will not always correspond to the
image name. In particular, if we start publishing both alpine and ubuntu images
we'll want the image name to have a suffix reflecting that.

spinnakerbot committed:
chore(dependencies): Autobump korkVersion (#619)

kevinawoo committed:
Update HtmlToPlainText.java

kevinawoo committed:
Merge branch 'master' into feature/dont-wordbreak-notifications

kevinawoo committed:
spotless........

kevinawoo commented:
https://github.com/spinnaker/echo/pull/616 is the real pr
